spec/services/hyrax/collections/migration_service_spec.rb

```diff
@@ -250,6 +341,84 @@
+      let!(:as_mg) { build(:no_solr_grants_adminset, id: 'as_mg', user: user, with_permission_template: { manage_groups: ['edit_group_1', 'edit_group_2'] }) }
+
+      before do
+        allow(AdminSet).to receive(:all).and_return([as_none, as_vu, as_vg, as_du, as_dg, as_mu, as_mg])
+      end
+
+      it 'sets read and edit access in solr doc' do # rubocop:disable RSpec/ExampleLength
+        AdminSet.all.each do |adminset|
+          expect(adminset.edit_users).to include(user.user_key)
+          expect(adminset.read_users).to eq []
+        end
+
+        Hyrax::Collections::MigrationService.repair_migrated_collections
+
+        AdminSet.all.each do |adminset|
+          pt_id = Hyrax::PermissionTemplate.find_by!(source_id: adminset.id)
+          expect(pt_id).not_to be_nil
+          expect_solr_access(pt_id, 'user', :manage, adminset.edit_users)
+          expect_solr_access(pt_id, 'group', :manage, adminset.edit_groups)
+          expect_solr_access(pt_id, 'user', :view, adminset.read_users)
+          expect_solr_access(pt_id, 'group', :view, adminset.read_groups)
+          expect_solr_access(pt_id, 'user', :view, adminset.read_users)
+          expect_solr_access(pt_id, 'group', :view, adminset.read_groups)
+        end
+      end
+    end
+
+    context 'when newer adminsets are found (e.g. adminsets created at or after Hyrax 2.1.0)' do
+      let!(:adminset) do
+        build(:adminset_lw,
+              id: 'as_newer', user: user,
+              with_permission_template: {
+                manage_users: [editor1.user_key, editor2.user_key],
+                deposit_users: [depositor1.user_key, depositor2.user_key],
+                view_users: [viewer1.user_key, viewer2.user_key],
+                manage_groups: ['manage_group_1', 'manage_group_2'],
+                deposit_groups: ['deposit_group_1', 'deposit_group_2'],
+                view_groups: ['view_group_1', 'view_group_2']
+              },
+              with_solr_document: true)
+      end
+      let!(:permission_template) { adminset.permission_template }
+      let!(:edit_users) { adminset.edit_users }
+
+      before do
+        allow(AdminSet).to receive(:all).and_return([adminset])
+      end
+
+      it "doesn't change the adminset" do # rubocop:disable RSpec/ExampleLength
+        pt_id = Hyrax::PermissionTemplate.find_by!(source_id: adminset.id).id
+        expect(pt_id).to eq permission_template.id
+        expect_solr_access(pt_id, 'user', :manage, adminset.edit_users)
+        expect_solr_access(pt_id, 'group', :manage, adminset.edit_groups)
+        expect_solr_access(pt_id, 'user', :view, adminset.read_users)
+        expect_solr_access(pt_id, 'group', :view, adminset.read_groups)
+        expect_solr_access(pt_id, 'user', :view, adminset.read_users)
+        expect_solr_access(pt_id, 'group', :view, adminset.read_groups)
```

Comment:
Here and right and lines 418-419 as well.

Reply:
Ugh, can't believe I missed these.  Thanks.